src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToStartHandler.java

```diff
@@ -1,56 +1,65 @@
 
+import static java.util.Objects.isNull;
 import static java.util.Objects.requireNonNull;
-import static org.apache.commons.collections4.CollectionUtils.emptyIfNull;
-import static uk.gov.hmcts.reform.sscs.model.AppConstants.POSTPONEMENTS_NOT_POSSIBLE_FOR_GAPS;
 
-import java.time.LocalDateTime;
-import java.util.Optional;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
 import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
 import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
-import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Hearing;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
 import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.sscs.util.SscsUtil;
 
 
 @Service
 public class PostponementRequestAboutToStartHandler implements PreSubmitCallbackHandler<SscsCaseData> {
 
+    public static final String NOT_LIST_ASSIST_CASE_ERROR = "Postponement requests can only be made for list"
+        + " assist cases";
+
+    @Value("${feature.snl.enabled}")
+    private boolean isScheduleListingEnabled;
+
     @Override
     public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
         requireNonNull(callback, "callback must not be null");
         requireNonNull(callbackType, "callbacktype must not be null");
 
-        return callbackType.equals(CallbackType.ABOUT_TO_START)
-                && callback.getEvent() == EventType.POSTPONEMENT_REQUEST
-                && callback.getCaseDetails() != null;
+        return callbackType == CallbackType.ABOUT_TO_START
+            && callback.getEvent() == EventType.POSTPONEMENT_REQUEST
+            && isScheduleListingEnabled;
     }
 
     @Override
     public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
         final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
         final SscsCaseData sscsCaseData = caseDetails.getCaseData();
 
-        final Optional<Hearing> optionalHearing = emptyIfNull(sscsCaseData.getHearings()).stream()
-                .filter(h -> h.getValue().getHearingDateTime().isAfter(LocalDateTime.now()))
-                .distinct()
-                .findFirst();
-
         final PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
 
-        if (HearingRoute.GAPS.equals(sscsCaseData.getRegionalProcessingCenter().getHearingRoute())) {
-            response.addError(POSTPONEMENTS_NOT_POSSIBLE_FOR_GAPS);
+        if (!SscsUtil.isSAndLCase(sscsCaseData)) {
+            response.addError(NOT_LIST_ASSIST_CASE_ERROR);
```

Comment:
The error code changed. The two errors emphasize the different semantics of the if statement.
The previous one stated that postponements are not possible for GAPS, the new one states that only list assist cases can have postponements.

- Recommend to check whether there were requirements around the error phrasing or alternatively restore the previous phrasing

Reply:
There are no requirements around the verbiage. We can use the new error message without problems.

Resolving this conversation.